tests/channels/test_http_handler.py

```diff
@@ -0,0 +1,288 @@
+        "GET",
+        "/graphql",
+    )
+    response = await client.get_response()
+    assert response == {
+        "headers": [b"Allow", b"GET, POST"],
+        "status": 405,
+        "body": b"Method not allowed",
+    }
+
+
+@pytest.mark.parametrize("consumer", [GraphQLHTTPConsumer, SyncGraphQLHTTPConsumer])
+@pytest.mark.parametrize("method", ["DELETE", "HEAD", "PUT", "PATCH"])
+async def test_disabled_methods(consumer, method: str):
+    client = HttpCommunicator(
+        consumer.as_asgi(schema=schema),
+        method,
+        "/graphql",
+        headers=[(b"accept", b"text/html")],
+    )
+    response = await client.get_response()
+    assert response == {
+        "headers": [b"Allow", b"GET, POST"],
+        "status": 405,
+        "body": b"Method not allowed",
+    }
+
+
+@pytest.mark.parametrize("consumer", [GraphQLHTTPConsumer, SyncGraphQLHTTPConsumer])
+async def test_fails_on_multipart_body(consumer):
+    client = HttpCommunicator(
+        consumer.as_asgi(schema=schema),
+        "POST",
+        "/graphql",
+        body=generate_body("{ hello }"),
+        headers=[(b"content-type", b"multipart/form-data")],
+    )
+    response = await client.get_response()
+    assert response == {
+        "status": 500,
+        "headers": [],
+        "body": b"Unable to parse the multipart body",
+    }
+
+
+@pytest.mark.parametrize("consumer", [GraphQLHTTPConsumer, SyncGraphQLHTTPConsumer])
+@pytest.mark.parametrize("body", [b"{}", b'{"foo": "bar"}'])
+async def test_fails_on_missing_query(consumer, body: bytes):
+    client = HttpCommunicator(
+        consumer.as_asgi(schema=schema),
+        "POST",
+        "/graphql",
+        body=body,
+    )
+    response = await client.get_response()
+    assert response == {
+        "status": 500,
+        "headers": [],
+        "body": b"No GraphQL query found in the request",
+    }
+
+
+@pytest.mark.parametrize("consumer", [GraphQLHTTPConsumer, SyncGraphQLHTTPConsumer])
+@pytest.mark.parametrize("body", [b"", b"definitely-not-json-string"])
+async def test_fails_on_invalid_query(consumer, body: bytes):
+    client = HttpCommunicator(
+        consumer.as_asgi(schema=schema),
+        "POST",
+        "/graphql",
+        body=body,
+    )
+    response = await client.get_response()
+    assert response == {
+        "status": 500,
+        "headers": [],
+        "body": b"Unable to parse request body as JSON",
+    }
+
+
+@pytest.mark.parametrize("consumer", [GraphQLHTTPConsumer, SyncGraphQLHTTPConsumer])
+async def test_graphql_post_query_fails_using_params(consumer):
+    client = HttpCommunicator(
+        consumer.as_asgi(schema=schema),
+        "GET",
+        "/graphql?foo=bar",
+    )
+    response = await client.get_response()
+    assert response == {
+        "status": 500,
+        "headers": [],
+        "body": b"No GraphQL query found in the request",
+    }
+
+
+# FIXME: All the tests bellow runs fine if running tests in this file only,
```

Comment:
Do we need to address this? 🤔

Reply:
I spent some time trying to understand this but did not get what is going on.

As the comment says, the tests run fine if I run only this file with pytest, but the whole testsuite makes this break with asyncio errors. So, something in the testsuite is probably affecting this.

Reply:
Have you tried adding `@pytest.mark.asyncio` decorators to all the tests?

Reply:
Just tried, didn't work T_T

Just so that you know, this is the issue that happens when running the whole testsuite:

```
============================================================================================== short test summary info ==============================================================================================
FAILED 😰  tests/channels/test_http_handler.py::test_graphql_query[SyncGraphQLHTTPConsumer] - asyncio.exceptions.TimeoutError
FAILED 😰  tests/channels/test_http_handler.py::test_graphql_can_pass_variables[SyncGraphQLHTTPConsumer] - asyncio.exceptions.TimeoutError
FAILED 😰  tests/channels/test_http_handler.py::test_graphql_get_query_using_params[SyncGraphQLHTTPConsumer] - asyncio.exceptions.TimeoutError
FAILED 😰  tests/channels/test_http_handler.py::test_graphql_can_pass_variables_using_params[SyncGraphQLHTTPConsumer] - asyncio.exceptions.TimeoutError
FAILED 😰  tests/channels/test_http_handler.py::test_returns_errors_and_data[SyncGraphQLHTTPConsumer] - asyncio.exceptions.TimeoutError
FAILED 😰  tests/channels/test_http_handler.py::test_graphql_get_does_not_allow_mutation[SyncGraphQLHTTPConsumer] - asyncio.exceptions.TimeoutError
FAILED 😰  tests/channels/test_http_handler.py::test_graphql_get_not_allowed[SyncGraphQLHTTPConsumer] - asyncio.exceptions.TimeoutError
```

The same code when running just `poetry run pytest tests/channels` (which runs not only this file, but the whole channels testsuite) also doesn't present the same issue.

That's why I suspect that something in another test is messing up here and not something related to the test itself. Decorating with `django_db` fixture worked around that though

Reply:
it works for me for the whole test suite 🤔